### DISCUSSION TITLE
how to lazy: plugins with submodules

### DISCUSSION THREAD

**Discussion by jaskerv** [nix-community/nixvim#4302] [General]
Hey all

I'm new to nix and nixvim. I'm trying to wrap my head around using `plugins.lazy.plugins`.

Can anyone show me how to use this with submodules ?
I got something like this atm
```
 plugins = with pkgs.vimPlugins; [
      {
        pkg = telescope-nvim;
	    lazy = true;
      }
      ./bufferline.nix
    ];
```

while bufferline.nix:
```
{ pkgs, ... }: {
  pkg = pkgs.vimPlugins.bufferline-nvim;
}
```

its telling me `error: attribute 'pkgs' missing`. Any help would be appreciated!

**Comment by MattSturgeon** [comment]
I think this is a misunderstanding of how submodules work.

While submodules _can_ be defined as module functions, `pkgs` is not an argument that's automatically available to submodules (more generally, this is true for any module-eval configuration).

Only `lib`, `options`, and `config` are automatically provided by the module system; all other args come from `_module.args` or `specialArgs`.

In this case, `pkgs` is added to the top-level _nixvim_ module configuration's args by our nixpkgs module. Or if you're using home-manager/NixOS it is also added to _that_ module configuration's args by _their_ nixpkgs module.

However, the `plugins.lazy.plugins.*` submodule does _not_ have a `pkgs` module arg.

---

As an asside, is there a particular reason you're using the lazy.nvim module instead of nixvim's more integrated `plugins.*.lazyLoad` feature? See https://nix-community.github.io/nixvim/user-guide/lazy-loading.html
